Lib/test/test_dis.py

```diff
@@ -2462,66 +2462,70 @@ def test_invokation(self):
             ('-S', '--specialized'),
         ]
 
-        infile = self.infile('def f():\n\tprint(x)\n\treturn None')
+        self.set_source('''
+            def f():
+                print(x)
+                return None
+        ''')
 
         for r in range(1, len(base_flags) + 1):
             for choices in itertools.combinations(base_flags, r=r):
                 for args in itertools.product(*choices):
                     with self.subTest(args=args[1:]):
-                        _ = self.invoke_dis(infile, *args)
+                        _ = self.invoke_dis(*args)
 
     def test_show_cache(self):
         # test 'python -m dis -C/--show-caches'
         source = 'print()'
-        expect = '''\
-0           RESUME                   0
-
-1           LOAD_NAME                0 (print)
-            PUSH_NULL
-            CALL                     0
-            CACHE                    0 (counter: 0)
-            CACHE                    0 (func_version: 0)
-            CACHE                    0
-            POP_TOP
-            LOAD_CONST               0 (None)
-            RETURN_VALUE
-'''
+        expect = textwrap.dedent('''
```

Comment:
`set_source` does the decent. Do we need it in all these places as well?

Reply:
No you're right. I don't know why I didn't update it.

Reply:
I eventually added a utility function that normalizes the strings and I call it whenever needed.